core/platform/auth/firebase_auth_services.py

```diff
@@ -96,7 +103,10 @@ def establish_firebase_connection():
             raise
 
 
-def establish_auth_session(request, response):
+def establish_auth_session(
+        request: webapp2.Request,
+        response: webapp2.Response
+) -> None:
```

Comment:
```suggestion
def establish_auth_session(
        request: webapp2.Request,
        response: webapp2.Response) -> None:
```

Prefer keeping closing parenthesis on the same line (unless the return type annotation causes it to exceed 80 chars). Ditto for other places too.

cc @vojtechjelinek, @DubeySandeep for comments on this. My suggestion is based on what we have been doing elsewhere in the codebase since we don't have strict rules around this?

@DubeySandeep do you think the current styling is wrong? Should the closing parenthesis be right indented by 4 spaces?

Reply:
@sajalasati In our codebase, we are consistent in placing the return type in next line.

Reply:
Personally, I like the sad-face approach (some arguments why are in this [blog-post](https://lukasz.langa.pl/1d1a43c4-9c8a-4c5f-a366-7f22ce6a49fc/) except for the consistency argument). I think it is also useful because if the type is too long (`really_long_variable: ReallyLongType) -> ReallyLongType:`) then it makes sense to split it up before the `)`.

Reply:
Yes I agree with the advantages, and perhaps following a single rule will help. Thanks!